src/components/Courses.jsx

```diff
@@ -33,23 +34,19 @@ const Course = () => {
                             </Accordion>
                         ))
                     }
+                    {courses.length === 0 &&(
+                        <Card>
+                            <Card.Body>
+                                <Card.Body>no courses found...</Card.Body>
+                            </Card.Body>
+                        </Card>
+                    )}
                 </Row>
             </Container>  
         )
     }
     else {
-        return(
-            <Container>
-                <Row>
-                    <Card>
-                        <Card.Body>
-                            <Card.Title>Loading Courses</Card.Title>
-                            <Card.Text>Please wait...</Card.Text>
-                        </Card.Body>
-                    </Card>
-                </Row>
-            </Container>
-        )
+        return <Loading />
```

Comment:
It will be much better for readability, if you ditch the `if/else` and put this above as an early return.

e.g. before your main `return` add this:
```js
if (loading) {
    return <Loading />
}
````